### ISSUE TITLE
AJAX pagination

### ISSUE THREAD

**Issue by sobrinho** [grid-js/gridjs#40]
**Is your feature request related to a problem? Please describe.**
The dataset is too big to be fetched at once.

**Describe the solution you'd like**
It would be nice to have a defined API for remote pagination (probably for remote searching as well since the search would have to be done by the server).

**Describe alternatives you've considered**
Implement the pagination by myself but it would lose the point of this library having everything done.

**Additional context**
N/A

**Comment by amatzen** [comment]
I would love to see this being implemented aswell!

**Comment by afshinm** [comment]
Makes sense! I'm currently working on this feature and will release a version soon. thanks folks!

**Comment by lucashutyler** [comment]
If possible, it would be nice to be able to load a couple extra pages on either side of the current page (precaching?).  That way the user won't have to wait for the table to load precached items if they are slowly just going to next and previous pages.  They would still have to wait if they filtered or jumped to the end of the list, of course.  I'd imagine a parameter called precachedPages that would load that number of pages ahead and behind the current page.
